crates/signer/src/manager/dirk.rs

```diff
@@ -0,0 +1,917 @@
+                .connect()
+                .await
+            {
+                Ok(ch) => {
+                    channels.insert(domain, ch);
+                }
+                Err(e) => {
+                    trace!("Couldn't connect to Dirk with domain {}: {e}", domain);
+                }
+            }
+        }
+
+        let mut accounts: HashMap<String, Account> = HashMap::new();
+
+        for host in config.hosts {
+            let domain = host.domain.unwrap_or_default();
+            let channel = channels
+                .get(&domain)
+                .ok_or(eyre::eyre!("Couldn't connect to Dirk with domain {domain}"))?
+                .clone();
+
+            let (dirk_accounts, dirk_distributed_accounts) = get_accounts_in_wallets(
+                channel.clone(),
+                host.accounts
+                    .iter()
+                    .filter_map(|account| Some(account.split_once("/")?.0.to_string()))
+                    .collect(),
+            )
+            .await?;
+
+            for account_name in host.accounts.clone() {
+                let (wallet, _) = account_name.split_once("/").ok_or(eyre::eyre!(
+                    "Invalid account name: {account_name}. It must be in format wallet/account"
+                ))?;
+
+                // Handle simple accounts
+                for dirk_account in dirk_accounts.iter().filter(|a| {
+                    a.name == account_name || a.name.starts_with(&format!("{}/", account_name))
+                }) {
+                    let public_key = BlsPublicKey::try_from(dirk_account.public_key.as_slice())?;
+                    let key_name =
+                        dirk_account.name.split_once("/").map(|(_, n)| n).unwrap_or_default();
+                    trace!(?dirk_account.name, "Adding account to hashmap");
+                    let is_proxy = key_name.contains('/');
+
+                    accounts.insert(hex::encode(public_key), Account {
+                        wallet: wallet.to_string(),
+                        name: key_name.to_string(),
+                        public_key: Some(public_key),
+                        hosts: vec![HostInfo { domain: domain.clone(), participant_id: 1 }],
+                        wallet_type: WalletType::Simple,
+                        signing_threshold: 1,
+                        is_proxy,
+                    });
+                }
+
+                // Handle distributed accounts
+                for dist_account in dirk_distributed_accounts.iter().filter(|a| {
+                    a.name == account_name || a.name.starts_with(&format!("{}/", account_name))
+                }) {
+                    let public_key =
+                        BlsPublicKey::try_from(dist_account.composite_public_key.as_slice())?;
+                    let key_name =
+                        dist_account.name.split_once("/").map(|(_, n)| n).unwrap_or_default();
+                    let is_proxy = key_name.contains('/');
```

Comment:
Same here

Reply:
Should be done in 51fe16dcb1b307f9a0b6450ad3ad3ee62a7cfcda but I want to test it a bit more just in case.